geant4@10.02.p02 pulls wrong version of qt

`geant4@10.02.p02` requires `qt4` not `qt5`, see

```
     24    -- Using VecGeom install configuration at /cvmfs/oasis.opensciencegrid.org/osg/spack/opt/spack/linux-rhel7-x86_64/gcc-5.3.0/vecgeom-0.3.rc-pwnbkcxok4xg6kvf65c64vpheuqykuzx/lib/CMake/USolids/USolidsConfig.cmake.
  >> 25    CMake Error at /cvmfs/oasis.opensciencegrid.org/osg/spack/opt/spack/linux-rhel7-x86_64/gcc-5.3.0/cmake-3.10.0-f57tuxwjk7p64pq2cfyivrcf7ijcxwzi/share/cmake-3.10/Modules/FindQt4.cmake:1320 (message):
     26      Found unsuitable Qt version "5.9.1" from
     27      /cvmfs/oasis.opensciencegrid.org/osg/spack/opt/spack/linux-rhel7-x86_64/gcc-5.3.0/qt-5.9.1-jsi7xb357uzy36g5rhnrjxqggpcrudse/bin/qmake,
     28      this code requires Qt 4.x
     29    Call Stack (most recent call first):
     30      cmake/Modules/Geant4InterfaceOptions.cmake:117 (find_package)
     31      CMakeLists.txt:80 (include)
```

Need better restrictions:
```
[centos@cvmfs-rhel7-build-machine ~]$ spack spec geant4
Input spec
--------------------------------
geant4

Concretized
--------------------------------
geant4@10.02.p02%gcc@5.3.0 build_type=RelWithDebInfo +qt arch=linux-rhel7-x86_64
[...]
    ^qt@5.9.1%gcc@5.3.0~dbus~examples~gtk~krellpatch~opengl patches=7f34d48d2faaa108dc3fcc47187af1ccd1d37ee0f931b42597b820f03a99864c,c52f72dac7fdff5a296467536cc9ea024d78f94b49903286395f53fd0eb66e5e ~phonon~webkit arch=linux-rhel7-x86_64
[...]
```

--- a/var/spack/repos/builtin/packages/geant4/package.py
+++ b/var/spack/repos/builtin/packages/geant4/package.py
@@ -40,7 +40,7 @@ class Geant4(CMakePackage):
     version('10.02.p01', 'b81f7082a15f6a34b720b6f15c6289cfe4ddbbbdcef0dc52719f71fac95f7f1c')
     version('10.01.p03', '4fb4175cc0dabcd517443fbdccd97439')
 
-    variant('qt', default=True, description='Enable Qt support')
+    variant('qt', default=False, description='Enable Qt support')
 
     depends_on('cmake@3.5:', type='build')
 
@@ -51,7 +51,7 @@ class Geant4(CMakePackage):
     depends_on("zlib")
     depends_on("vecgeom")
     depends_on("xerces-c")
-    depends_on("qt@4.8:", when="+qt")
+    depends_on("qt@4.8:4.999", when="+qt")
 
     def cmake_args(self):
         spec = self.spec

